#!/usr/bin/env python
PACKAGE = "neat_movement"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Speed limits
gen.add("max_linear_vel_forward", double_t, 0, "Maximum forward speed (m/s)", 0.4, 0.0, 2.0)
gen.add("max_linear_vel_backward", double_t, 0, "Maximum backward speed (m/s)", 0.2, 0.0, 2.0)
gen.add("max_angular_vel", double_t, 0, "Maximum angular velocity (rad/s)", 0.3, 0.0, 3.0)

# Acceleration limits
gen.add("max_linear_accel", double_t, 0, "Maximum linear acceleration (m/s^2)", 0.05, 0.01, 2.0)
gen.add("max_linear_decel", double_t, 0, "Maximum linear deceleration (m/s^2)", 0.05, 0.01, 2.0)
gen.add("max_angular_accel", double_t, 0, "Maximum angular acceleration (rad/s^2)", 0.05, 0.01, 5.0)

# Safety zones (distances from robot center)
gen.add("decel_zone_distance", double_t, 0, "Deceleration zone distance (m)", 1.5, 0.0, 5.0)
gen.add("stop_zone_distance", double_t, 0, "Stop zone distance (m)", 0.5, 0.0, 3.0)
gen.add("decel_zone_speed_ratio", double_t, 0, "Speed ratio in decel zone (0-1)", 0.5, 0.0, 1.0)

# Robot dimensions
gen.add("robot_width", double_t, 0, "Robot width (m)", 1.0, 0.1, 5.0)
gen.add("robot_length", double_t, 0, "Robot length (m)", 1.0, 0.1, 5.0)

# Control parameters
gen.add("rotation_tolerance", double_t, 0, "Rotation angle tolerance (rad)", 0.02, 0.01, 0.5)
gen.add("xy_goal_tolerance", double_t, 0, "XY goal tolerance (m)", 0.03, 0.01, 1.0)
gen.add("rotation_kp", double_t, 0, "Rotation P gain", 2.0, 0.0, 10.0)
gen.add("rotation_kd", double_t, 0, "Rotation D gain", 0.5, 0.0, 5.0)

# Loop rate
gen.add("control_frequency", double_t, 0, "Control loop frequency (Hz)", 20.0, 1.0, 100.0)

exit(gen.generate(PACKAGE, "neat_movement", "SimpleMoveParams"))
